spec/models/services/facebook_spec.rb

```diff
@@ -54,7 +82,18 @@
 
     it "should include post url in message with photos" do
       post_params = @service.create_post_params(@status_message)
-      expect(post_params[:message]).to include 'http'
+      expect(post_params[:message]).to include '(via http:'
+    end
+
+    it "should include post url in message with photos when no text message" do
+      @status_message = alice.build_post(:status_message, :text => "")
+      @status_message.photos << @photos
+
+      @status_message.save!
+      alice.add_to_streams(@status_message, alice.aspects)
+
+      post_params = @service.create_post_params(@status_message)
+      expect(post_params[:message]).to include '(via http:'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.